### DISCUSSION TITLE
Handling the error on abort

### DISCUSSION THREAD

**Discussion by BrendanDougan** [katspaugh/wavesurfer.js#3313] [Q&A] [answered]
Is it possible to catch the error on abort, or alternatively not throw an error on abort?

**Comment by katspaugh** [comment]
`load` returns a promise, so if there’s a fetch error, you catch it. You have to use the load(url) method in this case and not the url option.

**Reply by BrendanDougan** [reply]
Hi, thanks for the quick response. I've modified the basic example in the codebase as below, and it's still throwing the error. The setTimeout is there to simulate a button click slightly later, which is the scenario in my code. It doesn't throw the error without the setTimeout. Any help appreciated.

```
// A basic example

import WaveSurfer from 'https://unpkg.com/wavesurfer.js@7/dist/wavesurfer.esm.js'

const controller = new AbortController();
const signal = controller.signal

const wavesurfer = WaveSurfer.create({
  container: document.body,
  waveColor: 'rgb(200, 0, 200)',
  progressColor: 'rgb(100, 0, 100)',
  fetchParams: { signal }
})

wavesurfer.on('destroy', () => {
  controller.abort();
})

wavesurfer.load( 'https://storage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4' ).catch(error =>
  console.log(error)
);

setTimeout( () => {
  wavesurfer.destroy();
}, 500)

wavesurfer.on('click', () => {
  wavesurfer.play()
})
```

**Reply by katspaugh** [reply]
Ah, I see, thank you. It's a bug in the fetch progress handler. It clones the response, so the main response is caught correctly (your console log works), but the second one throws. I'll fix it tomorrow.

**Reply by BrendanDougan** [reply]
That would be great, thanks.

**Reply by katspaugh** [reply, marked as answer]
Released a fix in 7.4.4. Thanks for the report!

**Reply by BrendanDougan** [reply]
That's great, thanks for the quick turnaround!
